### ISSUE TITLE
All eclipse instances show the same workspace label on the dock icon

### ISSUE THREAD

**Issue by banjaxxer** [turesheim/eclipse-launcher#5299]
Eclipse instances (even from different copies of eclipse in the Applications folder) launch with the same workspace label on the dock icon.  You can open preferences and then update the workspace name, but on the next launch the update will be lost.
Behaviour started in Mojave and continues in Catalina.

**Comment by banjaxxer** [comment]
This may be an eclipse bug because the workspace name is overwritten in a workspace not containing the plugin...

**Comment by turesheim** [comment]
Are you still having this issue? I'm unable to reproduce it.

**Comment by banjaxxer** [comment]
Yes still have it.  The workspace name in Eclipse preferences is the same, but the workspace path is correct in each instance.  I keep the workspace name and the last folder in the path the same.  So I have ~/workspaces/bob and ~/workspaces/alice but both show bob.   One of the Eclipse instances is 4.6 Oxygen 3a and the other is 4.8.0Photon June 2018.  I can update the workspace name while they are open but the name for 4.8 always wins when I re-open them again.

**Comment by noscito** [comment]
+1
